app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -613,6 +622,13 @@ public function canUseBolt()
      */
     private function isAllowedOnCurrentPageByRoute()
     {
+        $isAllowed = false;
```

Comment:
This is the wrong place for this code.  You are basically corrupting and overriding a well defined method with something completely unrelated.  This code should be removed from here and added to a private method as described above.